File: 594. Longest Harmonious Subsequence/optimised.ts
function findLHS(nums: number[]): number {
	const n = nums.length;
	const f = new Map<number, number>();
	for (const num of nums) {
		f.set(num, (f.get(num) || 0) + 1);
	}

	let ans = 0;
	for (const [num, count] of f) {
		if (f.has(num + 1)) {
			ans = Math.max(ans, count + f.get(num + 1));
		}
	}

	return ans
};

console.log(findLHS([1, 2, 5, 6, 6]));

File: 594. Longest Harmonious Subsequence/optimised.py
#!/usr/bin/env python

from typing import List

def findLHS(nums: List[int]) -> int:
    freq = {}
    for num in nums:
        if num in freq:
            freq[num] += 1
        else:
            freq[num] = 1

    ans = 0
    for num in freq:
        if num + 1 in freq:
            ans = max(ans, freq[num] + freq[num + 1])

    return ans

print(findLHS([1,3,2,2,5,2,3,7]))
